### ISSUE TITLE
Update old version to become most recent version.

### ISSUE THREAD

**Issue by NakedFool** [share/sharedb#387]

**Comment by alecgibson** [comment]
@NakedFool I think you're misunderstanding the purpose of `fetchSnapshot`. That method will only give you a read-only copy of the snapshot. It doesn't affect your current `doc.data` (which you should be able to see by inspection).

If you want to roll back to an editable copy of an old version, how you do so is left completely up to you. You have a number of options:

1. Create a brand new document with the content of the old snapshot. This is the simplest option, and would be a bit like "forking" the doc. You could potentially include some metadata in your middleware or something to create a link to an old doc if you need that sort of information.
2. Delete your current document and then create a new one with the same ID, and the old snapshot content. This is the simplest option if you also want to keep previous history
3. Diff the old snapshot with your current snapshot, and then apply the inverse operation. This is a bit like a "revert", and would be the most complex, but most semantic way of handling things

**Comment by happy15** [comment]
what about apply a new op with empty path to replace the whole json document with the specified old version?

**Comment by alecgibson** [comment]
@happy15 yes this is also a potential approach with types like `json0`, which support just overwriting your whole `data` object. This is pretty similar to just deleting it.
